File: task_16_4.py
class Time:
    SECONDS_PER_DAY = 24 * 60 * 60

    def __init__(self, hours=0, minutes=0, seconds=0):
        total = hours * 3600 + minutes * 60 + seconds
        self._set_from_seconds(total)

    def _set_from_seconds(self, total):
        total %= self.SECONDS_PER_DAY
        self.hours = total // 3600
        self.minutes = (total % 3600) // 60
        self.seconds = total % 60

    def to_seconds(self):
        return self.hours * 3600 + self.minutes * 60 + self.seconds

    def difference_in_seconds(self, other):
        return abs(self.to_seconds() - other.to_seconds())

    def add_seconds(self, value):
        return Time(seconds=self.to_seconds() + value)

    def add_minutes(self, value):
        return self.add_seconds(value * 60)

    def __str__(self):
        return f"{self.hours:02d}:{self.minutes:02d}:{self.seconds:02d}"


def main():
    h1, m1, s1 = map(int, input("Введите первое время (часы минуты секунды): ").split())
    h2, m2, s2 = map(int, input("Введите второе время (часы минуты секунды): ").split())

    first = Time(h1, m1, s1)
    second = Time(h2, m2, s2)

    print(f"Разница: {first.difference_in_seconds(second)} секунд")

    seconds = int(input("Сколько секунд прибавить к первому времени: "))
    print(f"Результат: {first.add_seconds(seconds)}")

    minutes = int(input("Сколько минут прибавить к первому времени: "))
    print(f"Результат: {first.add_minutes(minutes)}")


if __name__ == "__main__":
    main()

File: task_16_4.cpp
#include <iostream>
#include <iomanip>
#include <sstream>
#include <stdexcept>
#include <cstdlib>

using namespace std;

class Time {
private:
    int hours;
    int minutes;
    int seconds;

    static long long normalize(long long total) {
        const long long day = 24LL * 60 * 60;
        total %= day;
        if (total < 0) {
            total += day;
        }
        return total;
    }

    void fromSeconds(long long total) {
        total = normalize(total);
        hours = static_cast<int>(total / 3600);
        minutes = static_cast<int>((total % 3600) / 60);
        seconds = static_cast<int>(total % 60);
    }

public:
    Time(int h = 0, int m = 0, int s = 0) {
        if (h < 0 || h > 23 || m < 0 || m > 59 || s < 0 || s > 59) {
            throw invalid_argument("Некорректное время");
        }
        hours = h;
        minutes = m;
        seconds = s;
    }

    long long toSeconds() const {
        return static_cast<long long>(hours) * 3600 + minutes * 60 + seconds;
    }

    long long differenceInSeconds(const Time& other) const {
        return llabs(toSeconds() - other.toSeconds());
    }

    Time addSeconds(long long value) const {
        Time result;
        result.fromSeconds(toSeconds() + value);
        return result;
    }

    Time addMinutes(long long value) const {
        return addSeconds(value * 60);
    }

    string toString() const {
        ostringstream out;
        out << setfill('0') << setw(2) << hours << ':'
            << setw(2) << minutes << ':'
            << setw(2) << seconds;
        return out.str();
    }
};

#ifndef UNIT_TEST
int main() {
    try {
        int h1, m1, s1;
        int h2, m2, s2;

        cout << "Введите первое время (часы минуты секунды): ";
        cin >> h1 >> m1 >> s1;

        cout << "Введите второе время (часы минуты секунды): ";
        cin >> h2 >> m2 >> s2;

        Time first(h1, m1, s1);
        Time second(h2, m2, s2);

        cout << "Разность в секундах: " << first.differenceInSeconds(second) << '\n';

        long long value;
        cout << "На сколько секунд изменить первое время: ";
        cin >> value;
        cout << "Результат: " << first.addSeconds(value).toString() << '\n';

        cout << "На сколько минут изменить первое время: ";
        cin >> value;
        cout << "Результат: " << first.addMinutes(value).toString() << '\n';
    } catch (const exception& error) {
        cerr << error.what() << '\n';
        return 1;
    }

    return 0;
}
#endif
